tests/improver-apply-emos-coefficients/14-predictor_of_mean_with_mask.bats

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bats
+# -----------------------------------------------------------------------------
+# (C) British Crown Copyright 2017-2019 Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+. $IMPROVER_DIR/tests/lib/utils
+
+@test "apply-emos-coefficients using non-default predictor 'realizations'" {
+  improver_check_skip_acceptance
+  if python -c "import statsmodels" &> /dev/null; then
+      COEFFS="estimate-emos-coefficients/realizations/with_statsmodels_kgo.nc"
+      KGO="apply-emos-coefficients/masked/with_statsmodels_kgo.nc"
```

Comment:
This KGO file doesn't exist.

Reply:
Added.

I also had to recreate the with statsmodels KGO for test 04 (realizations/with_statsmodels_kgo.nc). This is broken in master (running master code against master acceptance test data). I recreated this KGO using this branch and then ran master IMPROVER against it, which passes. This demonstrates that this test is unchanged by the alterations in this branch, so this is just a fix that needs making and is now included in the KGO that will be copied across.

Reply:
BTW this can be tested by running the tests with /default-current or next rather than the production stack (which excludes statsmodels).

Reply:
Good plotting Stephen. It turns out the estimate-emos-coefficients KGO with statsmodels was incorrect. It was identical to the without statsmodels KGO.
So I have now also recreated the estimate-emos-coefficients/realizations/with_statsmodels_kgo.nc file using default-next, which does use statsmodels and gives a different answer to the without_statsmodels case. I then remade the test data for the apply-emos-coefficients tests 04 and 14 for the with statsmodels cases. I have run the 04 test again with master to check it still passes (14 doesn't exist in master), as well as all the estimate-emos-coefficients tests, all of these pass. So again, these are fixes to master rather than anything to do with this PR (test 14 data which is new and had to be remade in light of these corrections).

Reply:
I've queried the values in these KGO with Gavin as the difference the stats-model makes is ~3K which is bigger than I'd expect.

Reply:
Gavin has shown that the majority of the difference I saw is caused by the limit on the iteration count in the CLI test (in place to prevent the test taking a long time to run). When allowed to converge fully, the results show differences between "with" and "without" stats-models of no more than 0.8K.